src/main/java/org/maxgamer/quickshop/database/AbstractDatabaseCore.java

```diff
@@ -22,20 +22,34 @@
 import org.bukkit.plugin.Plugin;
 import org.jetbrains.annotations.NotNull;
 
-import java.sql.Connection;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
 
-public interface DatabaseCore {
+public abstract class AbstractDatabaseCore {
+    private final ReentrantLock lock = new ReentrantLock(true);
+    private final Condition conditionLock = lock.newCondition();
 
-    void close();
+    void waitForConnection() {
+        try {
+            lock.lock();
+            conditionLock.await();
+            lock.unlock();
+        } catch (InterruptedException ignored) {
+        }
+    }
 
-    void flush();
+    void signalForNewConnection() {
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/Ghost-chu/QuickShop-Reremake/pullRequest?prid=6383074)